src/schoology-extractor/schoology_extractor/mapping/attendance.py

```diff
@@ -115,7 +124,7 @@ def map_to_udm(attendance: list, section_associations: pd.DataFrame) -> pd.DataF
         ]
     ]
 
-    df["CreateDate"] = None
-    df["LastModifiedDate"] = None
+    if sync_callback is not None:
+        df = sync_callback(df)

```

Comment:
Something very important was missed before. Previously you were calling `sync_callback` but ignoring the result, so the output files did not have the `CreateDate` and `LastModifiedDate` columns.